includes/Core/Util/User_Input_Settings.php

```diff
@@ -126,19 +126,27 @@ function( $setting ) {
 	 * @return array|WP_Error User input settings.
 	 */
 	private function sync_with_proxy( $settings = null ) {
-		$user_input_settings_url  = $this->authentication->get_google_proxy()->url( Google_Proxy::USER_INPUT_SETTINGS_URI );
+		$user_input_settings_url = $this->authentication->get_google_proxy()->url( Google_Proxy::USER_INPUT_SETTINGS_URI );
+		$creds                   = $this->authentication->credentials()->get();
+
 		$user_input_settings_args = array(
 			'headers' => array(
 				'Authorization' => 'Bearer ' . $this->authentication->get_oauth_client()->get_access_token(),
+				'Content-Type'  => 'application/json',
+			),
+			'body'    => array(
+				'site_id'     => $creds['oauth2_client_id'],
+				'site_secret' => $creds['oauth2_client_secret'],
 			),
 		);
 
 		if ( ! empty( $settings ) ) {
-			$user_input_settings_args['headers']['Content-Type'] = 'application/json';
-			$user_input_settings_args['body']                    = wp_json_encode( $settings );
+			$user_input_settings_args['body']['settings']       = $settings;
+			$user_input_settings_args['body']['client_user_id'] = get_current_user_id();
```

Comment:
The settings endpoint currently expects `client_user_id` as a string (because we plan to support non-WordPress systems that may not use an integer user id). To make this work, cast the ID as a string. cc: @felixarntz please confirm that sounds right? The type sent must match what the endpoint expects or we get a service side "unmarshaling" error.
```suggestion
			$user_input_settings_args['body']['client_user_id'] = (string) get_current_user_id();
```

Reply:
Yes, that's correct. I missed that before because I was thinking of URL-encoded parameters where it would become a string anyways. But yeah I guess we need to cast this to a string here, although it's a bit unintuitive. Let's go with that for now, maybe in the future we can make the service more resilient to support an integer and cast it to a string on its own.

Reply:
Added.